File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        count = [0] * 10;

        for d in digits:
            count[d] +=1

        ans =0

        for num in range (100,1000,2):
            a = num // 100
            b = (num //10)%10
            c = num % 10

            need = [0] *10
            need[a] += 1
            need[b] += 1
            need[c] += 1

            possible  = True

            for d in range(10):
                if need[d] > count[d]:
                    possible = False
                    break

            if possible: 
                    ans += 1

        return ans            

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.java
class Solution {
    public int totalNumbers(int[] digits) {
        int[] count = new int[10];

        for (int digit: digits){
            count[digit]++;
        }

        int ans =0;

        for(int num = 100; num <= 998; num +=2){
            int a = num/100;
            int b = (num/10) %10;
            int c = num%10;

            int [] need = new int [10];
            need[a]++;
            need[b]++;
            need[c]++;

            boolean possible = true;

            for(int digit = 0; digit <10; digit++){
                if(need[digit] > count[digit]){
                    possible = false;
                    break;
                }
            }
            if(possible){
                ans++;
            }
        }
        return ans;
    }
}